image-hd.c

```diff
@@ -233,8 +234,15 @@ static int hdimage_setup(struct image *image, cfg_t *cfg)
 
 	hd->align = cfg_getint_suffix(cfg, "align");
 	hd->partition_table = cfg_getbool(cfg, "partition-table");
+	hd->primary_partitions = cfg_getint(cfg, "primary-partitions");
 	hd->disksig = strtoul(cfg_getstr(cfg, "disk-signature"), NULL, 0);
 
+	if (hd->primary_partitions > 4) {
+		image_error(image, "too many primary partition (%i). must be "
+				"inferior to 4\n", hd->primary_partitions);
```

Comment:
Hmm, If the variable is called `primary-partitions` then is is confusing. I'm not sure I would count the `extended partition` as a `primary partition`. Either way the comment does not match the condition (less vs. less or equal).

I think it would be better to have a `extended-partition` option. `0` means automatic (like before) and 1 to 4 specify the explicit partition number for the extended partition.

Reply:
indeed, it sounds less confusing.